### PR TITLE
Prompt & show config

### PR DESCRIPTION
**Pull request by jakolehm**
image

### CODE REVIEW

File: lib/pharos/config.rb [hunk 1]
```diff
@@ -18,14 +18,14 @@ class Config < Dry::Struct
     constructor_type :schema
 
     attribute :hosts, Types::Coercible::Array.of(Pharos::Configuration::Host)
-    attribute :api, Pharos::Configuration::Api
     attribute :network, Pharos::Configuration::Network
-    attribute :cloud, Pharos::Configuration::Cloud
-    attribute :addons, Pharos::Types::Hash.default({})
+    attribute :kube_proxy, Pharos::Configuration::KubeProxy
```
Comment (jakolehm) [inline lib/pharos/config.rb]: Reordered for clarity in output.

File: lib/pharos/config_schema.rb [hunk 2]
```diff
@@ -5,6 +5,14 @@
 
 module Pharos
   class ConfigSchema
+    DEFAULT_DATA = {
+      'hosts' => [],
+      'api' => {},
+      'network' => {},
+      'authentication' => {},
+      'kube_proxy' => {}
+    }.freeze
```
Comment (jakolehm) [inline lib/pharos/config_schema.rb]: We need to find better way in future for this.. without placeholder values dry-struct does not set defaults to embedded structs.

File: lib/pharos/configuration/kube_proxy.rb [hunk 3]
```diff
@@ -5,7 +5,7 @@ module Configuration
     class KubeProxy < Dry::Struct
       constructor_type :schema
 
-      attribute :mode, Pharos::Types::String
+      attribute :mode, Pharos::Types::String.default('iptables')
```
Comment (jakolehm) [inline lib/pharos/configuration/kube_proxy.rb]: Added for clarity.

File: lib/pharos/up_command.rb [hunk 4]
```diff
@@ -113,6 +119,32 @@ def configure(config)
...
+      puts pastel.green("==> Using following software versions:")
+      Pharos::Phases.components_for_config(config).sort_by(&:name).each do |c|
+        puts "    #{c.name}: #{c.version}"
+      end
+    end
+
+    # @param config [Pharos::Config]
+    def prompt_continue(config)
+      lexer = Rouge::Lexers::YAML.new
+      yaml = JSON.parse(config.to_h.to_json).to_yaml
```
[the remark this answers is not in the document]
Comment (jnummelin) [reply lib/pharos/up_command.rb]: ok. I guess there isn't any more elegant way to achieve this, without taking in parts of Rails. Rails has `stringify_keys`, but taking in Rails for that is also an overkill. Looks just really weird. :)

[the remark this answers is not in the document]
Comment (jakolehm) [reply lib/pharos/up_command.rb]: We could hide this in `Pharos::Config`.

[the remark this answers is not in the document]
Comment (jnummelin) [reply lib/pharos/up_command.rb]: sounds like a good option

[the remark this answers is not in the document]
Comment (jakolehm) [reply lib/pharos/up_command.rb]: Done.

**Comment by kke** [comment]
Also I don't know how that looks on a 16-color terminal.

**Pull request merged**